### ISSUE TITLE
Return Value of oAuth.RequestToken(code) is null Always

### ISSUE THREAD

**Issue by empitekrishan** [InstaSharp/InstaSharp#115]
Hi
I am using InstaSharp for my ASP.Net MVC application. My problem is every time following code returns null. ( RequestToken(code) always return a OAuthResponse Instance with AccessToken = null , User = null)

`OAuthResponse oAuthResponse = await oAuth.RequestToken(code);`

So i can t get user id from oAuthResponse. here is my code.

```
public async Task<IList<Media>> GetInstagramMediaList(string clientId, string clientSecret, string returnUrl, string code)
        {
            List<Media> mediaResponseList = new List<Media>();
            InstagramConfig instagramConfig = new InstagramConfig(clientId, clientSecret, returnUrl, string.Empty);
            OAuth oAuth = new OAuth(instagramConfig);
            OAuthResponse oAuthResponse = await oAuth.RequestToken(code);

            Users users = new Users(instagramConfig, oAuthResponse);
            MediasResponse instaResponse = await users.Recent(oAuthResponse.User.Id.ToString());
            mediaResponseList.AddRange(instaResponse.Data);

            while (instaResponse.Pagination.NextMaxId != null)
            {
                instaResponse = await users.RecentSelf(maxId: instaResponse.Pagination.NextMaxId, minId: instaResponse.Pagination.NextMinId, count: null, maxTimestamp: null, minTimestamp: null);
                mediaResponseList.AddRange(instaResponse.Data);
            }

            return new List<Media>(mediaResponseList);
        }
```

i have written this code in business service. not in controller.
if any one can give a solution for this it will be a great help to me. 
Thanks in advance.

**Comment by awhitehouse104** [comment]
I needed this as well and I think I've fixed it (ignore the first commit reference, I can't seem to remove it). I'm getting [this](http://stackoverflow.com/questions/38258489/instagram-oauth-login-returning-matching-code-was-not-found-or-was-already-used) issue as well if I don't log out, but it seems unrelated.

**Comment by empitekrishan** [comment]
Hi @awhitehouse104  

Thank you for your massive help. 

Cheers mate

**Comment by opacitychris** [comment]
Thank you very much for the update @awhitehouse104 it solved for us as well.

**Comment by assassin316** [comment]
@awhitehouse104 

Hi, can you show me how to add in the Content Type into the request ??

**Comment by assassin316** [comment]
@awhitehouse104

Can you please explain further how you fixed this issue?? Maybe a code sample ?

**Comment by assassin316** [comment]
@awhitehouse104

Can you please explain further how you fixed this issue?? Maybe a code sample ?

@empitekrishan and @opacitychris

Were you able to fix this issue according to @awhitehouse104 comment?

**Comment by opacitychris** [comment]
@assassin316 I manually edited oauth.cs as seen here:
https://github.com/IntersectCreative/InstaSharp/commit/6268383f7460e053781bc1d25dff2a7021c0452c

**Comment by assassin316** [comment]
Ok, I'll make sure I have the latest and test the call again, thanks for the reply  :)

**Comment by assassin316** [comment]
@opacitychris 

Hi, I've tested with the following code and it returns null, is there something I'm doing wrong ?


		var scopes = new List<OAuth.Scope>();
		scopes.Add(InstaSharp.OAuth.Scope.Basic);

		var link = InstaSharp.OAuth.AuthLink(_config.OAuthUri + "authorize",
											_config.ClientId,
											_config.RedirectUri,
											scopes,
											InstaSharp.OAuth.ResponseType.Token);

		// add this code to the auth object
		var auth = new InstaSharp.OAuth(_config);

		// now we have to call back to instagram and include the code they gave us
		// along with our client secret
		var oauthResponse = await auth.RequestToken(link);


oauthResponse.AccessToken is always null  :\

**Comment by opacitychris** [comment]
I cant tell from that code you posted -- if you download this project and manually go into the instasharp source code, oauth.cs, is there a line about the urlencoded media type?

request.Content.Headers.ContentType.MediaType = "application/x-www-form-urlencoded";

Seen in the post here:
https://github.com/IntersectCreative/InstaSharp/commit/6268383f7460e053781bc1d25dff2a7021c0452c

**Comment by assassin316** [comment]
I actually took the code I needed and then debugged it with Fiddler (yes, the content type was set), I saved the Fiddler session and here is the error I received:

		HTTP/1.1 400 Bad Request
		Content-Language: en
		Expires: Sat, 01 Jan 2000 00:00:00 GMT
		Vary: Cookie, Accept-Language
		Pragma: no-cache
		Cache-Control: private, no-cache, no-store, must-revalidate
		Date: Mon, 03 Oct 2016 17:05:16 GMT
		Content-Type: application/json
		Set-Cookie: csrftoken=7Lg0p9fpu5CcMUuPFJDQyfljgYmDNKo0; expires=Mon, 02-Oct-2017 17:05:16 GMT; Max-Age=31449600; Path=/; secure
		Connection: keep-alive
		Content-Length: 114

		{"code": 400, "error_type": "OAuthException", "error_message": "Matching code was not found or was already used."}

**Comment by opacitychris** [comment]
Interesting.  Unfortunately im unfamiliar with that error, but somethings up.

It may be related to too many API calls?

http://stackoverflow.com/questions/29953203/instagram-oauth-returning-no-matching-code-found-on-one-server

**Comment by assassin316** [comment]
Woww, good find!  I'll keep trying to solve this and report back!  Thanks  :)

**Comment by assassin316** [comment]
Damn, i still get the same result on another device...   :\    I'll inspect this with Fiddler...

**Comment by assassin316** [comment]
Ok so just to double check, can anyone tell me if this url is correct ?? This is the url sent to the **RequestToken** function:

**https://api.instagram.com/oauth/authorize?client_id=<CLIENT_ID>&redirect_uri=<APP_LINK>/&response_type=token&scope=basic**

**Comment by assassin316** [comment]
I have an update on this issue which you can view here:   #120

**Comment by fujiy** [comment]
Fixed in 2.0.5
